src/libraries/Directory.Build.props

```diff
@@ -29,7 +29,7 @@
     <BuildTargetFramework Condition="'$(BuildTargetFramework)' == '' and '$(TargetFramework)' != ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', '(-[^;]+)', ''))</BuildTargetFramework>
     <!-- Build all .NET Framework configurations when net48 is passed in. This is for convenience. -->
     <AdditionalBuildTargetFrameworks Condition="'$(BuildTargetFramework)' == 'net48'">net45;net451;net452;net46;net461;net462;net47;net471;net472</AdditionalBuildTargetFrameworks>
-    <AdditionalBuildTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildAllProjects)' == 'true'">$(AdditionalBuildTargetFrameworks);netstandard2.0</AdditionalBuildTargetFrameworks>
+    <AdditionalBuildTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildAllProjects)' == 'true'">$(AdditionalBuildTargetFrameworks);net461;netstandard2.0</AdditionalBuildTargetFrameworks>
```

Comment:
IIRC (@ericstj likely has more context) we only support building .NETCoreApp current and .NETStandard builds when building from source. Why do we require .NET Framework assets to be built as well with the switch from pkgproj to csproj?

Reply:
We build the .NETFramework assets indirectly throught package projs. Sourcebuild produces packages in the vertical builds so we will need this config.

Reply:
I don't understand what you mean by that. Please elaborate

Reply:
Source build doesn't have an all config leg. It builds packages in a vertical build. During the build part, it builds the netcoreapp and netstandard assets. During pack task, it builds pkgprojs which have project references to the csprojs and that results in building all the config (which includes the .NETFramework config)

Reply:
> It builds packages in a vertical build.

If source build is running as part of a vertical build, I wouldn't expect any .NETFrameworks in projects to be built.

> During pack task, it builds pkgprojs which have project references to the csprojs and that results in building all the config

We filter project references to only build for the applicable `BuildTargetFramework`. Why doesn't it filter?

Reply:
Because its not being set, its value is ""

Reply:
Why is that so?

Reply:
this is the only place we set it and this wont get executed https://github.com/dotnet/runtime/blob/main/src/libraries/Directory.Build.props#L29

Reply:
Let me rephrase this discussion. I don't yet understand how changing the way nuget packages are product could result in an addition of compile time assets. How was it possible to not include .NET Framework assets in pkgprojs and why isn't that anymore possible with the NuGet Pack task on i.e. csprojs?

Reply:
> s. How was it possible to not include .NET Framework assets in pkgprojs and

pkgproj pack doesnt report missing files as and error and we didnt check the sourcebuild packages to see if they contain all the assets.

>  how changing the way nuget packages are product could result in an addition of compile time assets.

Nuget pack does report this as error if there are missing files 


> why isn't that anymore possible with the NuGet Pack task on i.e. csprojs

We might be able to suppress that, but i talked with @dseefeld and they are already building the netfx config in the sourcebuild repo with this  https://github.com/dotnet/source-build/blob/release/5.0/patches/runtime/0033-Build-net4-in-source-build.patch, so i think we should follow the same thing